File: 2366. Minimum Replacements to Sort the Array/Solution2366.cpp
#include<vector>

using namespace std;

class Solution2366 {
public:
    long long minimumReplacement(vector<int>& nums) {
        long count = 0L;
        int n = nums.size(), prev = (int) 1e9, k = 0;
        for (int i = n - 1; i >= 0; i--) {
            k = (nums[i] + prev - 1) / prev;
            prev = nums[i] / k;
            count += (long) k - 1;
        }
        return count;
    }
};
// ref https://leetcode.com/problems/minimum-replacements-to-sort-the-array/discuss/2388265/JavaC%2B%2BPython-One-Reverse-Pass

File: 2366. Minimum Replacements to Sort the Array/Solution2366.py
from typing import List


class Solution2366:
    def minimumReplacement(self, nums: List[int]) -> int:
        res, n, prev, k = 0, len(nums), 1000000000, 0
        for i in range(n-1,-1,-1):
            k = (nums[i] - 1 + prev) // prev
            prev = nums[i] // k
            res += k - 1
        return res
# ref https://leetcode.com/problems/minimum-replacements-to-sort-the-array/discuss/2388265/JavaC%2B%2BPython-One-Reverse-Pass